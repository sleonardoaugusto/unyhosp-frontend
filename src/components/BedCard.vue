<template>
  <div class="card col-md-3">
    <p class="card__name">{{ bed.name }}</p>
    <div v-if="hasPacient" class="card__preview">
      <p class="card__text-content">
        Paciente:
        <router-link :to="{name: 'attendance', params: {pacientId: bed.pacient.id}}">{{ bed.pacient.name }}</router-link>
      </p>
      <p class="card__text-content">
        Data:
        <span>{{ bed.date }}</span>
      </p>
      <p class="card__text-content">
        DiagnÃ³stico:
        <span>{{ bed.pacient.diagnostic }}</span>
      </p>
      <p class="card__text-content">
        Data:
        <span>{{ bed.pacient.health_plan }}</span>
      </p>
    </div>
    <div v-else>
      <p class="card__text-content">
        Vazio
      </p>
    </div>
  </div>
</template>

<script>
export default {
  name: 'BedCard',
  props: {
    bed: {
      type: Object,
      required: true,
      default: () => ({})
    }
  },
  computed: {
    hasPacient() {
      return this.bed.pacient
    }
  }
}
</script>

<style scoped>

</style>