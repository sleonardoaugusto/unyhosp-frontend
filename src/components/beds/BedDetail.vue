<template>
  <div>
    <h1>Bed Detail Page</h1>
  </div>
</template>

<script>
export default {
  name: 'BedDetail',
  props: ['bedId']
}
</script>

<style scoped>

</style>