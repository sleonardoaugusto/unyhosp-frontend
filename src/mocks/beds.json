{
  "data": [
    {
      "id": 1,
      "name": "Box A",
      "uti_id": 1,
      "date": "09/12/2019",
      "pacient": {
        "name": "Maria Joaquina",
        "diagnostic": "Apendicite",
        "health_plan": "UNIMED"
      }
    },
    {
      "id": 2,
      "name": "Box B",
      "uti_id": 1,
      "date": "02/12/2019",
      "pacient": {
        "name": "Pedro Alvares",
        "diagnostic": "Dengue",
        "health_plan": "Santa Casa"
      }
    },
    {
      "id": 3,
      "name": "Box C",
      "uti_id": 1,
      "date": null,
      "pacient": null
    }
  ]
}