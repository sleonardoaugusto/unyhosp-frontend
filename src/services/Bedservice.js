import apiClient from './config'

export default {
  get() {
    return apiClient.get('/beds/')
  },
  post(data) {
    return apiClient.post('/beds/', data)
  }
}