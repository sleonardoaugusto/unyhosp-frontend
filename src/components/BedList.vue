<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <bed-card v-for="(b, i) in beds" :bed="b" :key="i"/>
  </div>
</template>

<script>
import beds from '@/mocks/beds'
import BedCard from '@/components/BedCard'

export default {
  name: 'BedList',
  components: {
    BedCard
  },
  data: () => ({
    beds: []
  }),
  created() {
    this.beds = beds.data
  }
}
</script>

<style scoped>

</style>