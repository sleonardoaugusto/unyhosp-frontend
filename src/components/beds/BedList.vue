<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <bed-card v-for="(b, i) in beds" :bed="b" :key="i"/>
  </div>
</template>

<script>
import BedCard from '@/components/beds/BedCard'
import Bedservice from '@/services/Bedservice'

export default {
  name: 'BedList',
  components: {
    BedCard
  },
  data: () => ({
    beds: []
  }),
  created() {
    Bedservice.get()
      .then((res) => {
        this.beds = res.data
      })
  }
}
</script>

<style scoped>

</style>