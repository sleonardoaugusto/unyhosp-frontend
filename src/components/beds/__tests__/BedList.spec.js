import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import BedList from '@/components/beds/BedList'
import BedCard from '@/components/beds/BedCard'
import beds from '@/components/beds/__tests__/__mocks__/beds'
import Bedservice from '@/services/Bedservice'
import flushPromises from 'flush-promises'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

describe('BedList.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(BedList, {
      localVue
    })
  })

  it('component is rendering properly', () => {
    wrapper.setData({ beds: beds.data })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('not render child component if bed data is empty', () => {
    wrapper.setData({ beds: [] })
    const bedcard = wrapper.find(BedCard)
    expect(bedcard.exists()).toBeFalsy()
  })

  it('renders child component if bed data is not empty', () => {
    wrapper.setData({ beds: beds.data })
    const bedcard = wrapper.find(BedCard)
    expect(bedcard.exists()).toBeTruthy()
  })

  it('BedService get service method should be called on created lifecycle hook', async () => {
    Bedservice.get = jest.fn(() => Promise.resolve(beds))
    wrapper = shallowMount(BedList, { localVue, router })
    await flushPromises()
    expect(wrapper.vm.beds).toBe(beds.data)
  })
})