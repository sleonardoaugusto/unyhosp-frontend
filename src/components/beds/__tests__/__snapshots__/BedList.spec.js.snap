// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedList.vue component is rendering properly 1`] = `
<div class="col-md-12 d-flex justify-content-center flex-wrap">
  <bed-card-stub bed="[object Object]"></bed-card-stub>
  <bed-card-stub bed="[object Object]"></bed-card-stub>
  <bed-card-stub bed="[object Object]"></bed-card-stub>
</div>
`;
